tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+}
+
+func (r *ClusterTemplateTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+	r.templateConfig = new(ClusterTemplateTest)
+	config.LoadConfig("templateTest", r.templateConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+	r.client = client
+
+	enabled := true
+	var testuser = namegenerator.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	provider := provisioning.CreateProvider(r.templateConfig.TemplateProvider)
+	r.cloudCredentials, err = provider.CloudCredFunc(client)
+	require.NoError(r.T(), err)
+}
+
+func (r *ClusterTemplateTestSuite) TestProvisionRKE2TemplateCluster() {
+	_, err := r.client.Catalog.ClusterRepos().Create(context.TODO(), r.templateConfig.Repo, metav1.CreateOptions{})
+	require.NoError(r.T(), err)
+
+	watchRepo, err := r.client.Catalog.ClusterRepos().Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + r.templateConfig.Repo.Name,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	require.NoError(r.T(), err)
+
+	err = wait.WatchWait(watchRepo, func(event watch.Event) (ready bool, err error) {
+		if event.Type == watch.Added {
+			return true, nil
+		} else if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error waiting for repo to become active")
+		}
+		return false, nil
+	})
+	require.NoError(r.T(), err)
+
+	r.client.Session.RegisterCleanupFunc(func() error {
+		err := r.client.Catalog.ClusterRepos().Delete(context.TODO(), r.templateConfig.Repo.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+
+		watchRepo, err := r.client.Catalog.ClusterRepos().Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + r.templateConfig.Repo.Name,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = wait.WatchWait(watchRepo, func(event watch.Event) (ready bool, err error) {
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error while removing repository")
+			} else if event.Type == watch.Deleted {
+				return true, nil
+			}
+			return false, nil
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	latestVersion, err := r.client.Catalog.GetLatestChartVersion(r.templateConfig.TemplateName, r.templateConfig.Repo.ObjectMeta.Name)
+	require.NoError(r.T(), err)
+
+	project, err := projects.GetProjectByName(r.client, localCluster, "System")
+	require.NoError(r.T(), err)
+
+	installOptions := &charts.InstallOptions{
```

Comment:
Any reason why this is constructed here than in `InstallRKE2TemplateChart`, where I see `chartInstallActionPayload` being constructed anyways?

Reply:
Moving it would be fine. I think it does make a bit more sense to move it over.